grails-web-url-mappings/src/main/groovy/org/grails/web/mapping/DefaultUrlMappingEvaluator.java

```diff
@@ -502,7 +498,7 @@ private Object _invoke(String methodName, Object arg, Object delegate) {
                             urlDefiningMode = true;
                         }
                     }
-                } else if (!urlDefiningMode && CONSTRAINTS.equals(mappedURI)) {
+                } else if ((!urlDefiningMode || (parentResources.size() > 0 && parentResources.peek().isGroup)) && CONSTRAINTS.equals(mappedURI)) {
```

Comment:
Make sure we don't skip the constraints creation when underneath a group object.